sshd-common/src/main/java/org/apache/sshd/common/io/IoSession.java

```diff
@@ -82,6 +84,27 @@
      */
     Object removeAttribute(Object key);
 
+    /**
+     * Write a packet on the socket. Multiple writes can be issued concurrently and will be queued.
+     *
+     * @param  buffer      the buffer send. <B>NOTE:</B> the buffer must not be touched until the returned write future
+     *                     is completed.
+     * @return             An {@code IoWriteFuture} that can be used to check when the packet has actually been sent
+     * @throws IOException if an error occurred when sending the packet
+     */
+    IoWriteFuture writePacket(Buffer buffer) throws IOException;
+
+    /**
+     * Closes this session immediately or after all queued write requests are flushed. This operation is asynchronous.
+     * Wait for the returned {@link CloseFuture} if you want to wait for the session actually closed.
+     *
+     * @param  immediately {@code true} to close this session immediately. The pending write requests will simply be
+     *                     discarded. {@code false} to close this session after all queued write requests are flushed.
+     * @return             The generated {@link CloseFuture}
+     */
+    @Override
+    CloseFuture close(boolean immediately);
+
```

Comment:
Why is this override required ? Isn't the method inherited from `Closeable` enough?

Reply:
The javadoc is a bit more specific to the session, so I think it's worth overriding the method.

Reply:
Good enough...